src/drive/web/modules/upload/UploadQueue.jsx

```diff
@@ -29,45 +30,40 @@ const Pending = translate()(props => (
 ))
 
 const Item = translate()(({ file, status, isDirectory }) => {
+  const { CANCEL, LOADING, DONE_STATUSES, ERROR_STATUSES } = uploadStatus
   const { filename, extension } = splitFilename(file.name)
   let statusIcon
-  switch (status) {
-    case 'loading':
-      statusIcon = <Spinner class="u-ml-half" color={palette['dodgerBlue']} />
-      break
-    case 'cancel':
-      statusIcon = (
-        <Icon class="u-ml-half" icon="cross" color={palette['monza']} />
-      )
-      break
-    case 'failed':
-    case 'conflict':
-    case 'network':
-      statusIcon = (
-        <Icon class="u-ml-half" icon="warning" color={palette['monza']} />
-      )
-      break
-    case 'loaded':
-      statusIcon = (
-        <Icon
-          class="u-ml-half"
-          icon="check-circleless"
-          color={palette['emerald']}
-        />
-      )
-      break
-    case 'pending':
-    default:
-      statusIcon = <Pending />
-      break
+  let done = false
+  let error = false
+  if (status === LOADING) {
+    statusIcon = <Spinner class="u-ml-half" color={palette['dodgerBlue']} />
+  } else if (status === CANCEL) {
+    statusIcon = (
+      <Icon class="u-ml-half" icon="cross" color={palette['monza']} />
+    )
+  } else if (ERROR_STATUSES.includes(status)) {
+    error = true
+    statusIcon = (
+      <Icon class="u-ml-half" icon="warning" color={palette['monza']} />
+    )
```

Comment:
We'll have a warning icon for a few ms before we retry? Or do we skip it and put a new loading one? (just curious)

Reply:
Good point. I only see a loading icon even if I slow down the network. And in fact it makes sense since the status in redux never changes during the conflict management process.

Reply:
Thank for the clarification :)